hqmedia.upload_controller.js

```diff
@@ -286,6 +286,10 @@ function BaseHQMediaUploadController (uploader_name, marker, options) {
         if (!self.allowCloseDuringUpload) {
             self.allowClose = false;
         }
+        if (!self.isMultiFileUpload) {
+            var newExtension = '.' + self.filesInQueueUI[0].get('name').split('.').pop().toLowerCase();
```

Comment:
> What if there is more than one file in the queue? Should this be updated to handle each of them?
In bulk upload it looks like it's already copying the filepath over, so I don't think we will have this problem.

> What if there is no extension? I believe this will add a "." to the end of the filename in that case. Is that what we want? Maybe we don't care?
When you open up the modal to upload a video, it should show files that conform to the supported file types in vellum. The user can override that by selecting all files in the filetype dropdown, which would open us up to having no extension. MediaUploader also makes sure it's a valid video/image file after upload and doesn't change the file if it's invalid (i.e. I couldn't upload a jpg as a video when I tried locally).